Dockerfile

```diff
@@ -0,0 +1,58 @@
+# syntax=docker/dockerfile:1
+
+FROM condaforge/miniforge3:latest
+
+# Install system dependencies for biom-format and other scientific packages
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    libhdf5-dev \
+    zsh \
+    && rm -rf /var/lib/apt/lists/*
+
+# Create and activate mamba environment
+RUN mamba create -n toolenv python=3.13 r-base -y && \
+    echo "conda activate toolenv" >> /etc/profile.d/conda.sh
+
+# Set environment variables
+ENV CONDA_DEFAULT_ENV=toolenv
+ENV PATH=/opt/conda/envs/toolenv/bin:$PATH
+ENV PIP_ROOT_USER_ACTION=ignore
+
+# add conda shell integration to /root/.zshrc    
+RUN conda init zsh
+
+# Install Python packages with pip (recommended for conda env)
+RUN /opt/conda/envs/toolenv/bin/python -m pip install --upgrade pip --root-user-action=ignore && \
```

Comment:
all mamba statements as user, that way you can ignore "don't run as root" issues